addons/pos_sms/static/src/overrides/receipt_screen.js

```diff
@@ -5,35 +5,17 @@ import { _t } from "@web/core/l10n/translation";
 patch(ReceiptScreen.prototype, {
     setup() {
         super.setup(...arguments);
-        if (this.pos.config.module_pos_sms) {
-            this.state.input ||= this.currentOrder.get_partner()?.mobile || "";
-        }
-    },
-    isValidPhoneNumber(x) {
-        return x && /^\+?[()\d\s-.]{8,18}$/.test(x);
     },
-    actionSendReceipt() {
-        if (this.state.mode === "phone" && this.isValidPhoneNumber(this.state.input)) {
-            this.sendReceipt.call({ action: "action_sent_message_on_sms", name: "SMS" });
-        } else if (this.state.mode === "phone") {
+    actionSendReceiptOnSMS() {
+        if (this.isValidPhoneNumber(this.state.phone)) {
```

Comment:
the user not be able to click the button if the phone number is invalid anyways. We don't need 2 layers of protection here